pkg/plugin/backup_pvc_action_plugin.go

```diff
@@ -98,13 +99,11 @@ func (p *NewPVCBackupItemAction) Execute(item runtime.Unstructured, backup *vele
 	if err != nil {
 		return nil, nil, errors.WithStack(err)
 	}
-	if pv.Spec.PersistentVolumeSource.CSI == nil {
-		p.Log.Infof("Skipping PVC %s/%s, associated PV %s is not a CSI volume", pvc.Namespace, pvc.Name, pv.Name)
-		return item, nil, nil
-	}
-
-	if pv.Spec.PersistentVolumeSource.CSI.Driver != constants.VSphereCSIDriverName {
-		p.Log.Infof("Skipping PVC %s/%s, associated PV %s is not a vSphere CSI volume", pvc.Namespace, pvc.Name, pv.Name)
+	isVsphereCSIVolume := pv.Spec.PersistentVolumeSource.CSI != nil && pv.Spec.PersistentVolumeSource.CSI.Driver == constants.VSphereCSIDriverName
+	isMigratedVolume := pluginUtil.IsMigratedCSIVolume(pv)
+	shouldCreateSnapshotCR := isVsphereCSIVolume || (utils.IsFeatureEnabled(kubeClient, constants.CSIMigratedVolumeSupportFlag, false, p.Log) && isMigratedVolume)
+	if !shouldCreateSnapshotCR {
```

Comment:
The flow in previous code is much more readable, could you stick to it.

1. `if pv.Spec.PersistentVolumeSource.CSI == nil`
2. `if pv.Spec.PersistentVolumeSource.CSI.Driver != constants.VSphereCSIDriverName`
3. `if pluginUtil.IsMigratedCSIVolume(pv) {
          if !utils.IsFeatureEnabled(kubeClient, constants.CSIMigratedVolumeSupportFlag, false, p.Log
               {return item, nil, nil}
}`

Reply:
+1

Reply:
renamed shouldCreateSnapshotCR to isPVCBackUpSupported.

So the original structure was
"
if !isVsphereVolume {
   return
}

line 111-line200
"
We are adding an OR support, so cannot do early return to simplify the logic.

Reply:
@deepakkinni Please take an another look.